feature-wallet-impl/src/main/java/jp/co/soramitsu/wallet/impl/presentation/cross_chain/setup/CrossChainSetupViewModel.kt

```diff
@@ -643,6 +657,10 @@ class CrossChainSetupViewModel @Inject constructor(
 
     private fun setAddressByMetaAccountId(metaId: Long) {
         viewModelScope.launch {
+            val walletId = walletIdFlow.value
+            if (walletId != metaId) {
+                walletIdFlow.value = metaId
+            }
```

Comment:
In other words, you have implemented the "distinct" feature here? As I remember, the `walletIdFlow` is MutableStateFlow which already has the "distinct" under the hood.